extensions/escape-game/standalone.mjs

```diff
@@ -655,6 +661,74 @@ class Standalone {
     }
     return errorState;
   }
+
+
+  /**
+   * Give a redeem code to the user.
+   */
+   async s2s_sf_giveRedeemCode(srcExt, playerID, kwargs, sfInfo) {
+    const {nextState, errorState} = kwargs;
+    let {redeemCodes, eventName} = kwargs;
+
+    if (!(eventName in this.distributedCodes)) {
+      this.distributedCodes[eventName] = {};
+    }
+
+    let redeemCode = null;
+
+    if (!(playerID in this.distributedCodes[eventName])) {
+      for (let i = 0; i < redeemCodes.length; i++) {
```

Comment:
This currently takes O(N^2) because includes() takes O(N) time. Maybe make it O(N) or add a TODO?